go/rtl/auth_test.go

```diff
@@ -635,6 +635,65 @@ func TestAuthSession_Do_Timeout(t *testing.T) {
 			t.Errorf("Do() call did not error with context.DeadlineExceeded, got=%v", err)
 		}
 	})
+
+	t.Run("Do() follows Context set in AuthSession config", func(t *testing.T) {
```

Comment:
A bit of an ask, but could we test the overriding too?  (atleast partially)
So the first test case would only have config timeout set at 1 sec as it is now, then the next would have config timeout at 5 and option timeout 1, then next would config timeout 5, option timeout 5, and config context at 1 sec, and etc.

Reply:
Done.